packages/cw-orch-mock/src/core.rs

```diff
@@ -71,113 +70,268 @@ pub type MockApp = App<
 /// // We just use the MockState as an example here, but you can implement your own state struct.
 /// use cw_orch_mock::MockState as CustomState;
 ///
-/// let mock: Mock = Mock::new_custom("sender", CustomState::new());
+/// let mock: Mock = MockBase::new_custom("sender", CustomState::new());
 /// ```
-#[derive(Clone)]
-pub struct Mock<S: StateInterface = MockState> {
+pub struct MockBase<A: Api = MockApi, S: StateInterface = MockState> {
```

Comment:
Do we keep this as pub or make it private?

Reply:
We can make it private I think

Reply:
Let's leave it private inside the crate for deriving traits.